// @aws-sdk/rds-signer/dist-es/Signer.js
import { formatUrl } from "@aws-sdk/util-format-url";
import { HttpRequest } from "@smithy/protocol-http";
import { SignatureV4 } from "@smithy/signature-v4";
import { getRuntimeConfig as __getRuntimeConfig } from "./runtimeConfig";
export class Signer {
    constructor(configuration) {
        this.protocol = "https:";
        this.service = "rds-db";
        const runtimeConfiguration = __getRuntimeConfig(configuration);
        this.credentials = runtimeConfiguration.credentials;
        this.hostname = runtimeConfiguration.hostname;
        this.port = runtimeConfiguration.port;
        this.region = runtimeConfiguration.region;
        this.sha256 = runtimeConfiguration.sha256;
        this.username = runtimeConfiguration.username;
    }
    async getAuthToken() {
        const signer = new SignatureV4({
            service: this.service,
            region: this.region,
            credentials: this.credentials,
            sha256: this.sha256,
        });
        const request = new HttpRequest({
            method: "GET",
            protocol: this.protocol,
            hostname: this.hostname,
            port: this.port,
            query: {
                Action: "connect",
                DBUser: this.username,
            },
            headers: {
                host: `${this.hostname}:${this.port}`,
            },
        });
        const presigned = await signer.presign(request, {
            expiresIn: 900,
        });
        return formatUrl(presigned).replace(`${this.protocol}//`, "");
    }
}
